File: Data Structures & Algorithms/binary-search/submission-6.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        min = 0 
        max = len(nums) - 1 
        count = 0
        while(count < 15):
            i = (max - (min//2) - 1)
            if (len(nums) == 1):
                return 0
            if (nums[i] == target):
                return i
            elif (nums[i] < target):
                min = i
            elif(nums[i] > target):
                max = i
            count = count + 1
        return -1


        

File: Data Structures & Algorithms/two-integer-sum/submission-6.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }
        for(int i = 0; i < nums.length; i++){
            int num2 = target - nums[i];
            if (map.containsKey(num2)) {
                if (map.get(num2) != i) {

                
                    int[] solution = new int[2];
                    solution[0] = i;
                    solution[1 ]= map.get(num2);
                    return solution;
                }
            }
        }
        return null;
    }
}
